src/visGeometry/index.ts

```diff
@@ -121,31 +121,31 @@ class VisGeometry {
     public agentPaths: Map<number, AgentPath>;
     public mlogger: ILogger;
     // this is the threejs object that issues all the webgl calls
-    public threejsrenderer: WebGLRenderer;
-    public scene: Scene;
+    public threejsrenderer!: WebGLRenderer;
+    public scene!: Scene;
 
     public perspectiveCamera: PerspectiveCamera;
     public orthographicCamera: OrthographicCamera;
     public camera: PerspectiveCamera | OrthographicCamera;
     public controls: OrbitControls;
 
-    public dl: DirectionalLight;
-    public boundingBox: Box3;
-    public boundingBoxMesh: Box3Helper;
+    public dl!: DirectionalLight;
+    public hemiLight!: HemisphereLight;
+    public boundingBox!: Box3;
+    public boundingBoxMesh!: Box3Helper;
+    public tickMarksMesh!: LineSegments;
```

Comment:
This is kind of lame that typescript can't tell that we are initializing stuff inside of a sub-function called in the ctor. I'm assuming that's why you had to add the `!`s and that's almost certainly why the code was the way it was.